src/rest_client.cpp

```diff
@@ -9,7 +11,10 @@ rest_client::rest_client(std::string token) {
 }
 
 rest_client::~rest_client() {
-        // Wait for all HTTP threads to finish
-    while(http_thread_count > 0) {}
+    // Wait for all HTTP threads to finish
+    // Use sleep here so we don't do 100% cpu doing nothing but waiting. - Justasic
+    while(http_thread_count > 0)
+        std::this_thread::sleep_for(std::chrono::milliseconds(500));
```

Comment:
Thanks, good catch

Reply:
in C++14, this can be reduced to `std::this_thread::sleep_for(500ms)` as C++14 has pre-defined literals for time.